app/models/user/permissions.rb

```diff
@@ -12,6 +12,18 @@ module User::Permissions
     alias_attribute :allowed_service_ids, :member_permission_service_ids
   end
 
+  def permission_for_event?(event)
+    case event
+    when Cinstances::CinstancePlanChangedEvent, Cinstances::CinstanceCancellationEvent,
+           Cinstances::CinstanceExpiredTrialEvent, Applications::ApplicationCreatedEvent
+      has_permission?(:plans) && has_permission?(:partners) && has_access_to_service?(event.service.id)
+    when Services::ServiceDeletedEvent
+      has_access_to_service?(event.service_id)
+    else
+      true # To don't change the current behaviour for the rest, but we should end up ensuring the whole list from Notification.available_notifications
```

Comment:
This is the complete list:
```ruby
["application_created",
 "account_created",
 "limit_alert_reached_provider",
 "limit_violation_reached_provider",
 "account_plan_change_requested",
 "account_deleted",
 "cinstance_cancellation",
 "cinstance_plan_changed",
 "application_plan_change_requested",
 "service_contract_created",
 "service_plan_change_requested",
 "service_contract_plan_changed",
 "service_contract_cancellation",
 "cinstance_expired_trial",
 "invoices_to_review",
 "plan_downgraded",
 "expired_credit_card_provider",
 "unsuccessfully_charged_invoice_provider",
 "unsuccessfully_charged_invoice_final_provider",
 "message_received",
 "post_created",
 "csv_data_export",
 "daily_report",
 "weekly_report",
 "service_deleted"]
```
I need to know the permissions associated to most of them 😝

Reply:
I see that some of the info I need (although not all of it) is in:
https://github.com/3scale/porta/blob/76daf670e2d92c7253b2d191c05dcaf7c7f6368c/config/abilities/provider_member.rb#L19-L56

Reply:
I will create a new Jira issue about this.